backend/OKPluginVariableRendering/VariableRendering.cs

```diff
@@ -0,0 +1,358 @@
+                                {
+                                    var prio = 0;
+
+                                    switch (predicate[1..])
+                                    {
+                                        case "belongs_to_customer":
+                                            prio = customersGroupPrio;
+                                            break;
+                                        case "has_network_interface":
+                                            prio = networkInterfaceGroupPrio;
+                                            break;
+                                        case "is_attached_to":
+                                            prio = networkSegmentGroupPrio;
+                                            break;
+                                        case "is_assigned_to":
+                                            prio = moduleGroupPrio;
+                                            break;
+                                        case "belongs_to_assignment_group":
+                                            prio = assignmentGroupDefaultPrio;
+                                            break;
+                                        default:
+                                            break;
+                                    }
+
+                                    var a = new GatheredAttribute
+                                    {
+                                        SourceCIID = targetCI.ID,
+                                        Name = GetTargetName(attribute.Name, mapping.Target),
+                                        Value = attribute.Value,
+                                        RequiredTrait = follow.RequiredTrait,
+                                        Priority = prio,
+                                    };
+
+                                    if (tmpCIAttributes.ContainsKey(a.Name))
+                                    {
+                                        // check if current attribute has higher priority
+                                        if (tmpCIAttributes[a.Name].Priority < a.Priority)
+                                        {
+                                            tmpCIAttributes[a.Name] = a;
+                                        }
+                                    } else
+                                    {
+                                        tmpCIAttributes.Add(a.Name, a);
+                                    }
+
+                                    attribute.Name = a.Name;
+                                    attribute.Value = a.Value;
+                                }
+                            }
+                        }
+
+                        gatheredAttributes.AddRange(tmpCIAttributes.Values);
+                    }
+                }
+
+                // for all gathered attributes insert them to the main ci
+                foreach (var mapping in cfg.BaseCI.AttributeMapping)
```

Comment:
I don't understand what you are doing here, but it seems wrong... you should not loop through the attribute mappings here at all. `cfg.BaseCI.AttributeMapping` should only affect how attributes from the mainCI are mapped into gatheredAttributes.

Reply:
It looks like that this pas was misunderstood now I have changed the way how the attributes are gathered and I am using the ```cfg.BaseCI.AttributeMapping``` only for attributes of the base CI.